mantidproject/mantid
# Abins maintenance tweaks

**Description of work.**
In support of work to add features and improve performance of Abins, numerous refactoring and stylistic changes will emerge. As discussed in #26423 it is desirable to keep PRs small and focused, so this branch will be used for small batches of Abins tweaks that do not significantly impact functionality.

Work in progress: 
- I would like to make some more substantial changes to the caching system, which could hapen here but would be simpler to review as a separate PR based on top of this one.
- ~~The test I have added for FrequencyPowderGenerator would make more sense in _scripts/tests_, I should move it there before requesting that this PR is reviewed.~~

**To test:**
Run the standard unit tests. A new test has been added to AbinsIOmoduleTest relating to Python2/3 compatibility, so it is especially important to check this one with e.g.
```ctest -R AbinsIOmoduleTest --output-on-failure```

**Release notes**
*This does not require release notes* because there are no user-facing changes. Although I did rename the AbinsDaTa module to AbinsData which is arguably an API change...

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
###
